### ISSUE TITLE
[Bug Found and Fixed] MultiStepLR sometimes is useless

### ISSUE THREAD

**Issue by OuYaozhong** [Pointcept/Pointcept#379]
Hi, Pointcept Team, @Gofinge,

I found a bug while I develop my project based on your great repo.

It is relative to the Learning Rate Scheduler `MultiStepLR`.

### Bug Description
When you create the scheduler `MultiStepLR`, the milestones is defined as following:
[[source]](https://github.com/Pointcept/Pointcept/blob/014a2c6498dd3f4610f935ed8130d019e6389d49/pointcept/utils/scheduler.py#L27)
```
milestones=[rate * total_steps for rate in milestones],
```

The total step is relative to the `batch_size`, `epoch // eval_epoch`, and the length of `train dataset`, which are obviously.

However, in the source code of  `torch.optim.lr_scheduler.MultiStepLR`:
(
use the edition what I used as example
```
pytorch                   2.5.1           py3.11_cuda12.1_cudnn9.1.0_0    pytorch
```
)
[[source code of `torch.optim.lr_scheduler.MultiStepLR.get_lr`]](https://github.com/pytorch/pytorch/blob/a8d6afb511a69687bbb2b7e88a3cf67917e1697e/torch/optim/lr_scheduler.py#L593-L602)
```
def get_lr(self):
        """Compute the learning rate of each parameter group."""
        _warn_get_lr_called_within_step(self)

        if self.last_epoch not in self.milestones:
            return [group["lr"] for group in self.optimizer.param_groups]
        return [
            group["lr"] * self.gamma ** self.milestones[self.last_epoch]
            for group in self.optimizer.param_groups
        ]
```

In the `5-th` row (`if self.last_epoch not in self.milestones:`), the `self.milestones` is a `dict`, the keys of which is our defined milestones.

That means, only when our ` self.last_epoch` is in the `self.milestones`, the `MultiStepLR` will return a `lr = last_lr * self.gamma`.

However, the problem is, the key of `self.milestones`, (i.e., the milestones we input to `__init__`), will be a decimal, if the relationship between `epoch`, `eval_epoch`, `batch_size` and etc., is not integer multiple.

Meanwhile, the `self.last_epoch` is an integer in the class of PyTorch's `LRScheduler`, which will `+1` after `step` being executed. [[refer]](https://github.com/pytorch/pytorch/blob/a8d6afb511a69687bbb2b7e88a3cf67917e1697e/torch/optim/lr_scheduler.py#L237)

So, if the `milestones` are not integer, the `MultiStepLR` will not decrease the learning rate with multiple `gamma` at the corresponding `milestone`.

### Solution

Though I pay lot of time to find this bug, the solution is very simple.

Simply transfer the `milestones` input to the `__init__`  by force.

```
milestones=[int(rate * total_steps) for rate in milestones],
```

Done.

**Comment by Gofinge** [comment]
So sorry for my late response and thank you for finding the bug. Do you want to create a PR and I can merge this change? This way you can also be a contributor to Pointcept.

**Comment by OuYaozhong** [comment]
Hi, Pointcept Team and Gofinge, 

I have created a [PR  #399](https://github.com/Pointcept/Pointcept/pull/399#issue-2786509659).

It is my pleasure to have chance to contribute to this great project.
